sycl/include/sycl/ext/oneapi/experimental/builtins.hpp

```diff
@@ -117,6 +118,152 @@ inline __SYCL_ALWAYS_INLINE
 
 } // namespace native
 
+namespace detail {
+
+template <typename T> struct is_bf16_storage_type {
+  static constexpr int value = false;
+};
+
+template <> struct is_bf16_storage_type<uint16_t> {
+  static constexpr int value = true;
+};
+
+template <> struct is_bf16_storage_type<uint32_t> {
+  static constexpr int value = true;
+};
+
+template <int N> struct is_bf16_storage_type<vec<uint16_t, N>> {
+  static constexpr int value = true;
+};
+
+template <int N> struct is_bf16_storage_type<vec<uint32_t, N>> {
+  static constexpr int value = true;
+};
```

Comment:
And why such obtuse code? Just use templated variables like
```c++
template <typename T> constexpr auto is_bf16_storage_type = false;
template <> constexpr auto is_bf16_storage_type<std::uint32_t> = true;
```
etc.

Reply:
I don't know: this was added here: https://github.com/intel/llvm/pull/5748. I simply moved it to the correct experimental namespace in line with the extension document.  In any case it is deprecated already in favour of the `bfloat16` class implementation added in this PR and will be removed soon.